duckduckgo/iOS
# iOS 14 widgets and default browser

Task/Issue URL: 
* https://app.asana.com/0/414235014887631/1181475254517305
* https://app.asana.com/0/414235014887631/1189028203551370
Tech Design URL:
CC:

**Description**:

Adds default browser support and widgets for iOS 14

**Steps to test this PR**:

Using Xcode 12 and an iOS 14 beta simulator or device.  However, note that the profiles are required in order run default browser which means we have to use managed code signing (for now).  As such previously unregistered devices may not work.

Note: Integration tests disabled on BitRise for now, as simulator is not showing software keyboard.

***Default Browser***

1. Install the app, go to iOS Settings app > DuckDuckGo > Default Browser and choose DuckDuckGo
1. Launch URLs from other apps (e.g. calendar)
1. May get prompted to open in DuckDuckGo, allow this.

***Widgets***

1. Add each of the home screen widgets (long press home screen, hit the plus button)
1. Launch the app and add some favorites, ensure they show in the widgets
1. Edit the favorites, ensure the changes are reflected in the widgets
1. The small widget will launch a new search
1. The medium and large widget will launch whatever favorites you tap on, otherwise will launch a new search

***Regression***

1. Running on iOS 13 or earlier should show no difference in behaviour, but default browser and home screen widgets won't be available (obviously).



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
